modules/libpng/1.6.39/patches/add_build_file.patch

```diff
@@ -0,0 +1,84 @@
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,81 @@
++genrule(
++    name = "prebuilt_pnglibconf",
++    srcs = ["scripts/pnglibconf.h.prebuilt"],
++    outs = ["pnglibconf.h"],
++    cmd = "cp $< $@",
++)
++
++cc_library(
++    name = "libpng",
++    srcs = [
++        "png.c",
++        "pngerror.c",
++        "pngget.c",
++        "pngmem.c",
++        "pngpread.c",
++        "pngread.c",
++        "pngrio.c",
++        "pngrtran.c",
++        "pngrutil.c",
++        "pngset.c",
++        "pngtrans.c",
++        "pngwio.c",
++        "pngwrite.c",
++        "pngwtran.c",
++        "pngwutil.c",
++    ] + [
++        "pngconf.h",
++        "pngdebug.h",
++        "pnginfo.h",
++        "pngpriv.h",
++        "pngstruct.h",
++        ":prebuilt_pnglibconf",
++    ],
++    hdrs = ["png.h"],
++    visibility = ["//visibility:public"],
++    deps = ["@zlib"],
++)
++
++cc_test(
++    name = "pngtest_single_file",
++    size = "small",
++    srcs = ["pngtest.c"],
++    args = ["$(execpath pngtest.png)"],
```

Comment:
Since this path will be used at runtime, not during the build, `rootpath` should be used instead of `execpath`.

The Windows tests are failing since by default Bazel doesn't create the runfiles tree there. You can either disable the tests on Windows (e.g. with `target_compatible_with`) or wrap them in a shell test that uses `@bazel_tools//tools/bash/runfiles` to look up the correct path also on Windows.

Reply:
> rootpath should be used instead of execpath.

Done!

> The Windows tests are failing since by default Bazel doesn't create the runfiles tree there.

I took the disable approach, since I don't have a Windows machine. :crossed_fingers:, a Windows user can contribute a fix to allow these tests to work there.

Reply:
almost there -- you need a dep on `platforms`. alternatively just remove windows from presubmit.yml